### PR TITLE
LPS-68951 UserFinderImpl.countByC_FN_MN_LN_SN_EA_S is slow in case of using database instead index and having a lot of users

### PR DESCRIPTION
**Pull request by pei-jung**
https://issues.liferay.com/browse/LPS-68951

from @jorgediaz-lr: 

> Same solution than #149 but I have applied the changes requested by @shuyangzhou in shuyangzhou#3445 (comment)
> 
> In order to have two methods countByC_FN_MN_LN_SN_EA_S and findByC_FN_MN_LN_SN_EA_S with the specific SQL and not use a boolean flag, I have refactor original method:
> - commit 1: use paramsMapList instead having six variables params1...params6 39a0f04
> - commits 2 and 3: extract paramsMapList calculations to a new method that will be used by countByC_FN_MN_LN_SN_EA_S and findByC_FN_MN_LN_SN_EA_S cb308d0 beae391
> - commits 4, 5 and 6: minor code improvements a786861 96c2db0 780efa7
> 
> The code added in previous pull request is included in following commits:
> - commits 7, 8 and 9: add select count(userId) instead retrieving all users e36c4d0 3ea7fdd 3072533
> - commit 10: special code for sybase, similar to UserFinderImpl.countByGroups added in LPS-57748 (brianchandotcom#30280) eab18baf648c1af33950415b1d643921205094c9 

/cc @jorgeferrer @drewbrokke

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
Hi @shuyangzhou 

Do you have any update about this pull request?

Thank you!

**Comment by shuyangzhou** [comment]
Have not got a chance to check this yet. I will review it sometime this week.

**Comment by jorgediaz-lr** [comment]
Ok, thank you

**Comment by shuyangzhou** [comment]
@jorgediaz-lr Just talked to @pei-jung asking for a few changes. She will resend, thanks!

**Comment by jorgediaz-lr** [comment]
Hi @shuyangzhou 

Thank you for the information, I will wait for the changes

Regards

File: portal-impl/src/com/liferay/portal/service/persistence/impl/UserFinderImpl.java [hunk 1]
```diff
@@ -756,39 +1047,48 @@ else if (group.isUserGroup()) {
 						"usersOrgsTree",
 						new ArrayList<Organization>(organizations.values()));
 				}
+
+				paramsMapList.add(params2);
```
Comment (pei-jung) [inline portal-impl/src/com/liferay/portal/service/persistence/impl/UserFinderImpl.java]: @jorgediaz-lr 

For your reference, the paramsMapList is not populated correctly. We should only have one params2 in the list. On line 1112, if the condition passes, we need to override the params2 value that's being set here. params2 should not be added to the list here. With your changes, we can end up with two params2 in the list.

**Pull request closed without merging**